ChatSecure/Classes/Controllers/XMPP/OTRYapDatabaseRosterStorage.m

```diff
@@ -180,6 +180,11 @@ - (void)updateBuddy:(nullable OTRXMPPBuddy *)buddy withItem:(nonnull NSXMLElemen
     [[OTRDatabaseManager sharedInstance].readWriteDatabaseConnection asyncReadWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
         [newBuddy saveWithTransaction:transaction];
     }];
+    
+    if (buddy.pendingApproval && !newBuddy.pendingApproval) {
+        // Buddy was approved
+        [[NSNotificationCenter defaultCenter] postNotificationName:OTRBuddyPendingApprovalDidChangeNotification object:buddy userInfo:nil];
```

Comment:
@chrisballinger Oh so that's what you "filter" out with setting "object" in the addObserver call, I see. Thanks for pointing that out!